homeassistant/components/smarttub/config_flow.py

```diff
@@ -45,9 +51,35 @@ async def async_step_user(self, user_input=None):
             )
 
         existing_entry = await self.async_set_unique_id(account.id)
-        if existing_entry:
-            self.hass.config_entries.async_update_entry(existing_entry, data=user_input)
-            await self.hass.config_entries.async_reload(existing_entry.entry_id)
-            return self.async_abort(reason="reauth_successful")
+        if self._reauth_input is not None:
+            # this is a reauth attempt
+            if existing_entry:
+                if (
+                    existing_entry.unique_id
+                    != self._reauth_input[CONF_CONFIG_ENTRY].unique_id
+                ):
+                    # there is a config entry matching this account, but it is not the one we were trying to reauth
+                    return self.async_abort(reason="already_configured")
+                self.hass.config_entries.async_update_entry(
+                    existing_entry, data=user_input
+                )
+                await self.hass.config_entries.async_reload(existing_entry.entry_id)
+                return self.async_abort(reason="reauth_successful")
 
         return self.async_create_entry(title=user_input[CONF_EMAIL], data=user_input)
+
+    async def async_step_reauth(self, user_input=None):
+        """Get new credentials if the current ones don't work anymore."""
+        if user_input is not None:
+            self._reauth_input = dict(user_input)
```

Comment:
```suggestion
        self._reauth_entry = self.hass.config_entries.async_get_entry(self.context["entry_id"])
```

Reply:
```
    async def async_step_reauth(self, user_input=None):
        """Get new credentials if the current ones don't work anymore."""
        self._reauth_input = dict(user_input)
>       self._reauth_entry = self.hass.config_entries.async_get_entry(self.context["entry_id"])
E       KeyError: 'entry_id'
```

Reply:
The test isn't simulating with `raise ConfigEntryAuthFailed` does. I'll add a suggestion